### PR TITLE
Support reading arguments from config file and environment variables

### LINKED ISSUE #511: Feature request: configuration files
Having to configure exa using a shell alias is less than convenient. ripgrep allows you to set a RIPGREP_CONFIG_PATH environment variable to point to a file that contains a list of default options, one per line (https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#configuration-file). I think exa should do this too.

### PR DESCRIPTION
**Pull request by kidonng**
Idea and code adapted from [`bat`](https://github.com/sharkdp/bat).

Basically the same logic as bat:

- Look for `EXA_OPTS` environment variable
- If `EXA_OPTS` is not found, look for `EXA_CONFIG_PATH` and check if it exists
- Load arguments from `EXA_OPTS` or config file, and append the ones defined at runtime

Note that `--generate-config-file` is not implemented but I'd like to if there is any demand.

Doc is not updated for the moment because things may change 😄 

Resolves #511

I know little about Rust so correct me if I made any mistake 😃 (and I learned a lot from making this PR).

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -50,7 +50,13 @@ fn main() {
 
     logger::configure(env::var_os(vars::EXA_DEBUG));
 
-    let args: Vec<_> = env::args_os().skip(1).collect();
+    let cli_args = env::args_os().skip(1);
+    let args = get_args_from_env_var()
```
Comment (kidonng) [inline src/main.rs:54]: I'm not sure this is the most performant way, tried to use `append` or `extend` but no luck 😅

**Comment by jspiro** [comment]
Just installed exa and while I'm still unsure about the harder to type name than ls, the lack of config files like ripgrep and bat was a bit too pure. Thank you!

**Comment by ogham** [comment]
Hi, sorry about the delay on this one.

Adding a config file is something I do want to add, but I think exa could do with so many configuration options that merely having the config file be a list of command-line flags is going to be very limiting. I want to keep the option open to have a more structured configuration file.

**Comment by mfdorst** [comment]
I was really hoping some sort of config file was available, as I wanted to do `alias ls='exa -TL3'` but I realized that that would prevent me from specifying a different recursion depth would then be impossible. Is there some workaround for this in the meantime?

**Pull request closed without merging**